### PR TITLE
Remove unregistered types from supported target types

### PR DESCRIPTION
**Pull request by bison**
This checks for errors resulting from unregistered types when
initializing the MachineAutoscaler reconciler and removes them from
the list of supported target types.

### CODE REVIEW

**Comment by ingvagabund** [comment]
This might hide the case where a user forgets to deploy machineset crd. Worth documenting it somewhere. Or, maybe to make at least one of machineset or machinedeployment crds exists. If not, fail?

**Comment by bison** [comment]
> This might hide the case where a user forgets to deploy machineset crd. Worth documenting it somewhere. Or, maybe to make at least one of machineset or machinedeployment crds exists. If not, fail?

I don't want to try to do anything too clever, but we could check that there is at least one supported type found otherwise return an error. Does that sound like a reasonable compromise?

**Comment by ingvagabund** [comment]
> I don't want to try to do anything too clever, but we could check that there is at least one supported type found otherwise return an error. Does that sound like a reasonable compromise?

+1

File: pkg/controller/machineautoscaler/machineautoscaler_controller.go [hunk 1]
```diff
@@ -94,11 +99,23 @@ func (r *Reconciler) AddToManager(mgr manager.Manager) error {
 				ToRequests: handler.ToRequestsFunc(targetOwnerRequest),
 			})
 
-		if err != nil {
+		// If we get an error indicating that no matching type is registered
+		// with the API, then remove it from the list of supported target GVKs.
+		// If the type is later registered, a restart of the operator will pick
```
Comment (frobware) [inline pkg/controller/machineautoscaler/machineautoscaler_controller.go]: Does this have to be a forced restart (by a human)?

Comment (bison) [reply pkg/controller/machineautoscaler/machineautoscaler_controller.go]: Yes. For now. Seems reasonable to me. The immediate use case for this is just to not crash if one of the upstream or OpenShift groups is missing.

**Comment by enxebre** [comment]
I agree with @bison is out of the scope to deploy SupportedTargetGVKs list
/approve

**Pull request merged**